intermediate_source/reinforcement_q_learning.py

```diff
@@ -438,44 +368,38 @@ def optimize_model():
     # Optimize the model
     optimizer.zero_grad()
     loss.backward()
-    for param in policy_net.parameters():
-        param.grad.data.clamp_(-1, 1)
+    # In-place gradient clipping
+    torch.nn.utils.clip_grad_value_(policy_net.parameters(), 100)
     optimizer.step()
 
 
 ######################################################################
 #
 # Below, you can find the main training loop. At the beginning we reset
-# the environment and initialize the ``state`` Tensor. Then, we sample
-# an action, execute it, observe the next screen and the reward (always
+# the environment and obtain the initial ``state`` Tensor. Then, we sample
+# an action, execute it, observe the next state and the reward (always
 # 1), and optimize our model once. When the episode ends (our model
 # fails), we restart the loop.
 #
-# Below, `num_episodes` is set small. You should download
-# the notebook and run lot more epsiodes, such as 300+ for meaningful
-# duration improvements.
+# Below, `num_episodes` to 1000, but you should see the model constantly
+# achieve 500 steps within 600 training episodes.
 #
 
-num_episodes = 50
+num_episodes = 1000
```

Comment:
Did we measured how much it would take to finish compared to the original one?

Reply:
I guess that now that things are not pixel-based we can (slightly) recuce that, what do you think @pseudo-rnd-thoughts  @SiftingSands ?
But I don't think 50 episodes will be enough

Reply:
I vaguely remember 1000 episodes taking less than 10 minutes with the latest changes here (on a 2080ti). I want to say 50 episodes on the original one took 1-2 minutes, but that lead to obviously insufficient performance.